sdk/core/Azure.Core.TestFramework/README.md

```diff
@@ -67,8 +67,6 @@ public class AppConfigurationTestEnvironment : TestEnvironment
     public string Endpoint => GetRecordedVariable("APPCONFIGURATION_ENDPOINT");
     // Variables retrieved using GetVariable will not be recorded but the method will throw if the variable is not set
     public string SystemAssignedVault => GetVariable("IDENTITYTEST_TEST_SYSTEMASSIGNEDVAULT");
-    // Variables retrieved using GetOptionalVariable will not be recorded and the method will return null if variable is not set
```

Comment:
If we take this out, do we still have something that discusses sanitization in the template?

Reply:
Doesn't look like it. I could add a comment about sanitizers and matchers in the test base's ctor.

Reply:
There is almost no reason for partners to use GetOptionalVariable. Most of the existing usages are wrong. We shouldn't advertise it.

Reply:
What is the correct use case for `GetOptionalVariable`?

Reply:
To consume some environment variable that's set outside of normal New-TestResources flow. For example, it can be some password for identity manual tests. Most partners shouldn't have optional variables as outputs of test-resources.json are always available and they shouldn't be getting settings from any other source.

Reply:
All partners should think about is whether it's a secret or not. We may want to hide GetVariable as well (even if it means that we might record something that's not needed for the playback). It will simplify the mental model a lot.

Reply:
But most ppl will need to use sanitization, yes?  I'm fine taking the hard to use API out, but it would be good to make it clear to partners how to sanitize test logs

Reply:
@pakrym pointed out that we are doing this in the lines below this removal.  I withdraw my comment.

Reply:
We also have this comment in the template:

```csharp
        // You can sanitize secrets from recorded variables e.g., principal secrets.
        // See https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/core/Azure.Core.TestFramework#test-environment-and-live-test-resources for variables,
        // and https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/core/Azure.Core.TestFramework#sanitizing for sanitizing responses.
        public string KeyVaultSecret => GetRecordedVariable("KEYVAULT_SECRET", options => options.IsSecret());
```

Missed it when I was looking through for other info.